### PR TITLE
Feature/decimal answer format

### PR DESCRIPTION
**Pull request by snake984**

### CODE REVIEW

**Comment by coroutineDispatcher** [comment]
Hi. Thank you for your contribution in our code. Please, apply `ktlint` to your code and make a new commit. Other than that, it looks good :+1: 
Please send us a request in the top right corner of this page when you are finished so we can get notified to review as quick as possible.

**Comment by coroutineDispatcher** [comment]
Looks good to me.

File: survey/src/main/java/com.quickbirdstudios.surveykit/backend/views/question_parts/DoubleTextFieldPart.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    }
+
+    companion object {
+        fun withHint(context: Context, @StringRes hint: Int) = DoubleTextFieldPart(
+            context
+        ).apply {
+            field.hint = context.getString(hint)
+        }
+    }
+}
```
Comment (coroutineDispatcher) [inline survey/src/main/java/com.quickbirdstudios.surveykit/backend/views/question_parts/DoubleTextFieldPart.kt:24]: I think after applying `ktlint` there should be a new line here. Are you sure you applied it in all files?

File: survey/src/main/java/com.quickbirdstudios.surveykit/backend/views/questions/DoubleQuestionView.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    //region Private Helpers
+    private fun String.parseToDoubleOrNull(): Double? {
+        return try {
+            this.toDouble()
+        } catch (e: NumberFormatException) {
+            null
+        }
+    }
+    //endregion
+}
```
Comment (coroutineDispatcher) [inline survey/src/main/java/com.quickbirdstudios.surveykit/backend/views/questions/DoubleQuestionView.kt:63]: same

**Comment by snake984** [comment]
PR is updated, however I can seem to be able to add any reviewer

File: survey/src/main/java/com.quickbirdstudios.surveykit/result/question_results/DoubleQuestionResult.kt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import java.util.Date
+
+@Parcelize
+data class DoubleQuestionResult(
+    override val id: Identifier,
+    override val startDate: Date,
+    override var endDate: Date = Date(),
+    override val stringIdentifier: String,
+    val answer: Double?
+) : QuestionResult
```
Comment (coroutineDispatcher) [inline survey/src/main/java/com.quickbirdstudios.surveykit/result/question_results/DoubleQuestionResult.kt:15]: same

File: survey/src/main/java/com.quickbirdstudios.surveykit/backend/views/question_parts/DoubleTextFieldPart.kt (hunk 1 above)
Comment (snake984) [reply survey/src/main/java/com.quickbirdstudios.surveykit/backend/views/question_parts/DoubleTextFieldPart.kt:24]: Capture d’écran, le 2020-08-11 à 17 32 04

Ok this is weird, the empty lines at the end of files are still here in my code (I opened the file with TextEdit and it's still here) but it looks like Github got rid of them. I ran Ktlint on all the project.

Comment (snake984) [reply survey/src/main/java/com.quickbirdstudios.surveykit/backend/views/question_parts/DoubleTextFieldPart.kt:24]: Oh sorry I see what you meant by applying it, not juste the lint, the format

**Comment by snake984** [comment]
I applied Ktlint on the whole code, it should be good. However I don't see why the empty lines are not showing on github...

File: survey/src/main/java/com.quickbirdstudios.surveykit/backend/views/questions/DoubleQuestionView.kt [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        questionAnswerView.field.gravity = Gravity.CENTER
+        questionAnswerView.field.hint = answerFormat.hint
+        questionAnswerView.field.afterTextChanged { footer.canContinue = isValidInput() }
+        val preselected = preselected?.toString() ?: answerFormat.defaultValue?.toString()
+        questionAnswerView.field.setText(preselected ?: context.getString(R.string.empty))
+    }
+    //endregion
+
+    //region Private Helpers
+    private fun String.parseToDoubleOrNull(): Double? {
```
Comment (KlausNie) [inline survey/src/main/java/com.quickbirdstudios.surveykit/backend/views/questions/DoubleQuestionView.kt:55]: No need for that: `kotlin.String.toDoubleOrNull` already exists in the stdlib

File: survey/src/main/java/com.quickbirdstudios.surveykit/steps/QuestionStep.kt [hunk 5]
```diff
@@ -2,13 +2,49 @@ package com.quickbirdstudios.surveykit.steps
...
+import com.quickbirdstudios.surveykit.result.question_results.DoubleQuestionResult
+import com.quickbirdstudios.surveykit.result.question_results.EmailQuestionResult
+import com.quickbirdstudios.surveykit.result.question_results.ImageSelectorResult
+import com.quickbirdstudios.surveykit.result.question_results.IntegerQuestionResult
+import com.quickbirdstudios.surveykit.result.question_results.MultipleChoiceQuestionResult
+import com.quickbirdstudios.surveykit.result.question_results.ScaleQuestionResult
+import com.quickbirdstudios.surveykit.result.question_results.SingleChoiceQuestionResult
+import com.quickbirdstudios.surveykit.result.question_results.TextQuestionResult
+import com.quickbirdstudios.surveykit.result.question_results.TimeQuestionResult
+import com.quickbirdstudios.surveykit.result.question_results.ValuePickerQuestionResult
```
Comment (KlausNie) [inline survey/src/main/java/com.quickbirdstudios.surveykit/steps/QuestionStep.kt:47]: Please revert the import expansion. It's not really a big deal, it's just about keeping it the same throughout the project.

**Review by KlausNie — CHANGES_REQUESTED** [review]
2 small things open, then we can merge 👍

**Pull request merged**